src/OpenTelemetry.Api/Trace/TracerProvider.cs

```diff
@@ -50,5 +51,44 @@ public Tracer GetTracer(string name, string version = null)
 
             return new Tracer(new ActivitySource(name, version));
         }
+
+        /// <summary>
+        /// Adds instrumentation source.
+        /// </summary>
+        /// <param name="source">DiagnosticSource name.</param>
+        /// <returns>TracerProvider instance.</returns>
+        public virtual TracerProvider AddSource(string source)
```

Comment:
Quick hack to extend public api as the logic is inside OpenTelemetry.dll

Reply:
A clever way to make it backward compatible.

I was thinking also of making `TracerProvider` an `abstract class`. Then the new methods could become abstract methods. However, this would be a kind of breaking change. As this would break the build if one would implement its own (custom) TraceProvider (anyone does it?).

Maybe we could make make `TracerProviderSdk` public and change `TracerProviderBuilderSdk.BuildSdk()` method to return `TracerProviderSdk` that would allow adding instrumentations at runtime?